lib/hcloud/entry_loader.rb

```diff
@@ -195,8 +196,30 @@ def _load(resource)
           next
         end
 
+        # if schema definition is [Class]
+        if definition.is_a?(Array) && definition.first.include?(EntryLoader)
+
+          # just set attribute to an empty array if value is no array or empty
+          if !value.is_a?(Array) || value.empty?
+            _update_attribute(key, [])
+            next
+          end
+
+          if value.first.is_a?(Integer)
+            # If value is an integer, this is the id of an object which's class can be 
+            # retreived from definition. The [] operator should automatically load the 
+            # appropriate object.
+            _update_attribute(key, value.map { |item| definition.first.new(client, id: item) })
```

Comment:
@tonobo AFAIK do I need to convert the definition class (e.g. Volume) to a resource class (VolumeResource), where I then can make use of the `[]`-operator to lazy load the resource by id. But there is no mapping. If I'm doing this like above, the volume object from my server will stay the same with he id initialized.

```
c = Hcloud::Client.new(token: "...")
c.servers.first.volumes.first
> c.servers.first.volumes.first.name
=> nil

Reply:
As discussed with @tonobo personally, a future object should be initialized instead of creating a concrete object of type `definition`. The Future class needs to know how to `find` the concrete object by `id`. Either by mapping the class name to the resource class/resource collection function in the client or with a lookup of the e.g. `Volume` to `VolumeResource`.